### PR TITLE
split api calls

### PR DESCRIPTION
**Pull request by LiorGnieslawMoveo**

### CODE REVIEW

File: src/app/components/history/history.component.ts [hunk 1]
```diff
@@ -4,32 +4,47 @@ import { PokemonService } from '../../services/pokemon.service';
...
 })
 
-export class HistoryComponent implements OnInit{
+export class HistoryComponent implements OnInit {
   searchHistory: string[] = [];
   isLoggedIn: boolean = false;
 
+  pokemons: Pokemon[] = [];
+
+
```
Comment (Karina-Moveo) [inline src/app/components/history/history.component.ts]: Remoce extra space

File: src/app/services/pokemon.service.ts [hunk 2]
```diff
@@ -11,7 +11,7 @@ import { CookieService } from 'ngx-cookie-service';
 })
 export class PokemonService {
   private apiUrl = 'https://pokeapi.co/api/v2/pokemon/';
-  private isLoggedInSubject: BehaviorSubject<boolean> = new BehaviorSubject<boolean>(this.cookieService.get('isLoggedIn')==='true');
+  private isLoggedInSubject: BehaviorSubject<boolean> = new BehaviorSubject<boolean>(this.cookieService.get('isLoggedIn') === 'true');
```
Comment (Karina-Moveo) [inline src/app/services/pokemon.service.ts:16]: No need to check equal to true

File: src/app/services/pokemon.service.ts [hunk 3]
```diff
@@ -21,42 +21,40 @@ export class PokemonService {
...
-              ...pokemon,
-              types: details[index].types,
-              abilities: details[index].abilities,
-              height: details[index].height,
-              weight: details[index].weight
-            }));
-          })
-        );
-      })
+        imageUrl: `https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/${this.extractPokemonIdFromUrl(pokemon.url)}.png`,
```
Comment (Karina-Moveo) [inline src/app/services/pokemon.service.ts]: move it out into a const

File: src/app/services/pokemon.service.ts [hunk 4]
```diff
@@ -21,42 +21,40 @@ export class PokemonService {
...
+  }
+
+  private mapPokemon(detail: any, url: string): Pokemon {
+    return {
+      name: detail.name,
+      types: detail.types.map((type: any) => type.type.name),
+      abilities: detail.abilities.map((ability: any) => ability.ability.name),
+      height: detail.height,
+      weight: detail.weight,
+      imageUrl: `https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/${this.extractPokemonIdFromUrl(url)}.png`,
```
Comment (Karina-Moveo) [inline src/app/services/pokemon.service.ts]: move into const

**Review by Karina-Moveo — CHANGES_REQUESTED** [review]

File: src/app/services/pokemon.service.ts (hunk 2 above)
Comment (LiorGnieslawMoveo) [reply src/app/services/pokemon.service.ts:16]: this.cookieService.get('isLoggedIn') returns a string so I need do this check to enter the subject a boolean value.

**Review by Karina-Moveo — APPROVED** [review]

**Pull request merged**